astropy/table/column.py

```diff
@@ -222,21 +222,25 @@ def _convert_sequence_data_to_array(data, dtype=None):
             category=FutureWarning,
             message=".*Promotion of numbers and bools to strings.*",
         )
+
+        has_unit = (
+            hasattr(data, "__len__")
```

Comment:
Minor nit, and pending the bigger questions, but the same check for length is done on l.200, so one might as well factor that out.

Reply:
done